Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -237,15 +294,16 @@ void FieldProbe::ComputeDiags (int step)
         amrex::IndexType const Bytype = By.ixType();
         amrex::IndexType const Bztype = Bz.ixType();
 
-        //defined for use in determining which CPU contains the particles
-        int probe_proc = -1;
-
         // loop over each particle
         // TODO: add OMP parallel as in PhysicalParticleContainer::Evolve
+
+        long numparticles = 0; // particles on this MPI rank
```

Comment:
We can do this at once:
```suggestion
        bool const only_valid = true;
        bool const only_local = true;
        auto const numparticles = m_probe.NumberOfParticlesAtLevel(lev, only_valid, only_local);
        m_data.resize(numparticles * noutputs);
```

Reply:
I rewrote the m_data functionality so that I never need to resize it. Made it a vector and I push values to it. I clear the m_data every time diags are run. (Not a problem for iteration method because those values are stored on the particle). I think this way is better because it doesn't need to be resized with different options for different geometries

Reply:
That's possible, and you can make this operation more efficient if you call `m_data.reserve()` after the `clear()` to the expected number of particles. Otherwise `.push_back` will constantly copy the memory of the vector into a larger space every time you add a new particle entry.
https://en.cppreference.com/w/cpp/container/vector/reserve

Reply:
The problem with reserving the data is that the local number of particles is fond via the counter which is counted during the loop. What do you think?